scene/resources/material.cpp

```diff
@@ -1129,8 +1135,35 @@ void BaseMaterial3D::_update_shader() {
 		code += "	samp+= texture(p_sampler,p_triplanar_pos.zy * vec2(-1.0,1.0)) * p_weights.x;\n";
 		code += "	return samp;\n";
 		code += "}\n";
+		code += "\n\n";
+	}
+
+	if (flags[FLAG_USE_TEXTURE_SMOOTH_PIXEL_FILTER]) {
+		code += "vec4 texture_point_smooth(sampler2D smp, vec2 uv, vec2 pixel_size) {\n";
```

Comment:
My dumb butt discovered StringBuilder exists [yesterday](https://github.com/godotengine/godot/pull/87363) and now I wonder how greatly it would benefit here... Just food for thought.

Reply:
Since this is mostly compile time strings this would best be done with R-strings, this has been discussed before and is tracked here (though specifically for particles):
* https://github.com/godotengine/godot/issues/85595

Reply:
O the efficiency!

Reply:
Done: https://github.com/godotengine/godot/pull/89267